src/classes/ExpectSObject.cls

```diff
@@ -0,0 +1,88 @@
+public class ExpectSObject extends Expect {
+    private static Boolean isSuccess = true;
```

Comment:
remove fields isSuccess isFail exceptionString.

They are all static which is bad news if you're asserting two different sobjects in the same test. Keeping mutable state like this opens the class up to bugs. It might need to happen at some point but I think the use case of asserting a dml operation throws (or not) shouldn't require fields